src/clj/harja/palvelin/integraatiot/vayla_rest/sampo_api.clj

```diff
@@ -0,0 +1,60 @@
+(ns harja.palvelin.integraatiot.vayla-rest.sampo-api
+  (:require [com.stuartsierra.component :as component]
+            [compojure.core :refer [POST]]
+            [taoensso.timbre :as log]
+            [harja.palvelin.asetukset :refer [ominaisuus-kaytossa?]]
+            [harja.palvelin.komponentit.http-palvelin :refer [julkaise-reitti poista-palvelut julkaise-palvelu]]
+            [harja.palvelin.integraatiot.sampo.tuonti :as tuonti]
+            [harja.palvelin.integraatiot.sampo.vienti :as vienti]
+            [harja.palvelin.integraatiot.api.tyokalut.xml-skeemat :as xml-skeemat]
+            [harja.palvelin.integraatiot.api.tyokalut.kutsukasittely :as kutsukasittely]
+            [harja.palvelin.tyokalut.ajastettu-tehtava :as ajastettu-tehtava]
+            [harja.kyselyt.maksuerat :as q-maksuerat]
+            [harja.palvelin.integraatiot.sampo.kasittely.maksuerat :as maksuerat]
+            [harja.palvelin.integraatiot.sampo.kasittely.kustannussuunnitelmat :as kustannussuunnitelmat]
+            [harja.pvm :as pvm]))
+
+(defprotocol Maksueralahetys
+  (laheta-maksuera-sampoon [this numero]))
+
+(defn tee-paivittainen-lahetys-tehtava [{:keys [db integraatioloki]} {:keys [paivittainen-lahetysaika] :as api-sampo-asetukset}]
+  (if paivittainen-lahetysaika
+    (do
+      (log/debug "Ajastetaan maksuerien ja kustannussuunnitelmien lähetys ajettavaksi joka päivä kello: "
+        paivittainen-lahetysaika)
+        (ajastettu-tehtava/ajasta-paivittain paivittainen-lahetysaika
+        (fn [_] (vienti/aja-paivittainen-api-lahetys db integraatioloki api-sampo-asetukset))))
+      (constantly nil)))
+
+(defrecord ApiSampo [api-sampo-asetukset]
+  component/Lifecycle
+  (start [{http :http-palvelin db :db integraatioloki :integraatioloki :as this}]
+    (log/debug "Käynnistetään SampoAPI-komponentti")
+    (when (ominaisuus-kaytossa? :api-sampo)
+      (julkaise-reitti
+        http :sampo-vastaanotto
+        (POST "/sampo/api/harja" request
+          (kutsukasittely/kasittele-sampo-kutsu db integraatioloki :sisaanluku
+            request xml-skeemat/+sampo-kutsu+
+            (fn [kutsun-parametrit kutsun-data kayttaja db tapahtuma-id]
```

Comment:
Käyttämättömiä parametreja. Pitäisikö käyttää vai poistaa?

Reply:
Oletin, että nämä on pakollisia, koska tätä oikeasti kutsutaan ihan muualta. Tässä vain välitetään tämä funkkari eteenpäin. Mutta, koska kutsukäsittelyssä on sampokutsuille oma käsittely, niin voin sen laittaa tekemään mitä haluan. Eli ylimääräiset parametrit poistettu tästä ja kutsukäsittelystä.